### PR TITLE
Ports Firing Range Magnet Control to TGUI

### PR DESCRIPTION
**Pull request by Arthri**
https://github.com/orgs/ParadiseSS13/projects/4/views/1?pane=issue&itemId=38641334

## What Does This PR Do
Ported the magnets control's UI in the Firing Range to TGUI. Code and Frequency are no longer configurable

Configuration is now broadcasted directly to the magnet. Turning one magnet on does not turn all magnets on anymore.

- Added constants(`MIN_CONTROLLER_SPEED`, `MAX_CONTROLLER_SPEED`, `MIN_ELECTRICITY_LEVEL`, `MAX_ELECTRICITY_LEVEL`, `MIN_MAGNETIC_FIELD`, `MAX_MAGNETIC_FIELD`, `MAX_PATH_LENGTH`), and converted 0/1 to `FALSE`/`TRUE` wherever appropriate
- Defined list of valid paths in JS and in DM(`n`, `s`, `e`, `w`, `c`, `r`)
- Added `find_magnet` proc to find a linked magnet by UID

## Why It's Good For The Game
Makes the magnet control easier to use.

## Images of changes
https://github.com/ParadiseSS13/Paradise/assets/41360489/1c419513-a647-4ecc-8d81-45eeca3588bd

## Testing
- Hopped into the firing range and played around with the settings

## Changelog
:cl: cdui
tweak: Magnet Control converted to TGUI
/:cl:

### CODE REVIEW

File: code/game/machinery/magnetic_module.dm [hunk 1]
```diff
@@ -217,109 +227,193 @@
...
+		if("toggle_magnet_power")
+			var/obj/machinery/magnetic_module/magnet = find_magnet(params["id"])
+			if(!magnet)
+				return
+			magnet?.Cmd("toggle-power")
+		if("set_electricity_level")
+			var/obj/machinery/magnetic_module/magnet = find_magnet(params["id"])
+			if(!magnet)
+				return
+			var/new_value = text2num(params["newValue"])
```
[the remark this answers is not in the document]
Comment (Sirryan2002) [reply code/game/machinery/magnetic_module.dm]: yes

File: code/game/machinery/magnetic_module.dm [hunk 2]
```diff
@@ -217,109 +227,188 @@
...
+				return
+			var/newElectricityLevel = text2num(params["electricityLevel"])
+			if(newElectricityLevel == null)
+				return
+			magnet.Cmd("set-electriclevel", clamp(newElectricityLevel, MIN_ELECTRICITY_LEVEL, MAX_ELECTRICITY_LEVEL))
+		if("set_magnetic_field")
+			var/obj/machinery/magnetic_module/magnet = find_magnet(params["id"])
+			if(!magnet)
+				return
+			var/newMagneticField = text2num(params["magneticField"])
```
Comment (Sirryan2002) [inline code/game/machinery/magnetic_module.dm:339]: make this use `snake_case` please

File: code/game/machinery/magnetic_module.dm [hunk 3]
```diff
@@ -217,109 +227,188 @@
...
+			if(newMagneticField == null)
```
Comment (Sirryan2002) [inline code/game/machinery/magnetic_module.dm:340]: please make all of these `== null` checks use `isnull()` instead

**Review by Sirryan2002 — APPROVED** [review]
just those last few things + deconflict your TGUI

**Review by S34NW — APPROVED** [review]
Deconflict for merge

**Comment by S34NW** [comment]
thanos-fine-ill-do-it-myself

**Pull request merged**